libslide/src/bignum/bignum_add.rs

```diff
@@ -0,0 +1,143 @@
+mod bignum_add {
+    use crate::bignum::Bignum;
+    use std::ops;
+
+    // @todo - figure out a swap method to handle lines 17-26
+    impl ops::Add for Bignum {
+        type Output = Bignum;
+
+        fn add(self, rhs: Bignum) -> Bignum {
+            // this assumes the signs are both positive for now
+            let mut carry: u8 = 0;
+            let mut lhs_size: usize = self.dec.iter().count();
+            let mut rhs_size: usize = rhs.dec.iter().count();
+            let mut res_dec: Vec<u8> = Vec::new();
+            let mut big_vec: Vec<u8>;
+
+            // HANDLE DECIMAL
+
+            // swap if neccessary to make lhs smaller. Initialize big vec as the bigger vector
+            if lhs_size > rhs_size {
+                let tmp: usize = lhs_size;
+                lhs_size = rhs_size;
+                rhs_size = tmp;
+                big_vec = self.dec.clone();
+            } else {
+                big_vec = rhs.dec.clone();
+            }
```

Comment:
You could use mem::swap (import std::mem):
```suggestion
            if lhs_size > rhs_size {
                mem::swap(&lhs_size, rhs_size);
                big_vec = self.dec.clone();
            } else {
                big_vec = rhs.dec.clone();
            }
```

Reply:
see I read this was slow is the only thing so I wasn't sure